### PR TITLE
feat(sdk): bucket.add_object

### PR DESCRIPTION
**Pull request by Chriscbr**
Code written from the third Wingly Update twitch show. Still need to write unit tests and probably at least one integration test before shipping it.

I promise to get back to this PR just like GCP bucket (#827) 😆 

*By submitting this pull request, I confirm that my contribution is made under the terms of the 
[Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingsdk/src/target-tf-aws/bucket.ts
Comment (marciocadev) [inline libs/wingsdk/src/target-tf-aws/bucket.ts:110]: @Chriscbr Restrictions on object names in S3 are quite broad, but there are characters that need special treatment and others to avoid as per this link

https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html

Perhaps we can ignore these limitations for the moment (I imagine very few people want to create a file with @ or $) or we can change the resource-names so that it is not limited to resources but to everything related to names

And I imagine that this can also affect the logical id of the object

File: libs/wingsdk/src/target-tf-azure/bucket.ts
Comment (marciocadev) [inline libs/wingsdk/src/target-tf-azure/bucket.ts:106]: I don't know much about CdkTF, but I imagine there are limits on the length of the logical id string

File: libs/wingsdk/src/cloud/bucket.ts
Comment (hasanaburayyan) [inline libs/wingsdk/src/cloud/bucket.ts:42]: I did not see one but I wonder if we have any issue that references future support of blobs. Maybe a TODO: with the issue link might mitigate questions from those who wonder why we are using just string for body. 🤷

**Review by hasanaburayyan — APPROVED** [review]
❤️ the azure support as well. Small question on whether to add a TODO comment but other than that LGTM.

File: libs/wingsdk/src/cloud/bucket.ts
Comment (marciocadev) [reply libs/wingsdk/src/cloud/bucket.ts:42]: I had the same question, but I thought that since the language didn't have blob support yet that's why str was used

File: libs/wingsdk/src/target-tf-aws/bucket.ts
Comment (Chriscbr) [reply libs/wingsdk/src/target-tf-aws/bucket.ts:110]: Logical IDs don't have any hard restrictions on characters or length AFAIK.

Based on the docs you linked, maybe we could offer some warnings if users choose file names with special characters, but I think it's OK to keep things simple for now. For the size of the string, I think every cloud provider has some restrictions on user input sizes, but since the key name is entirely user-provided, I think it's OK to let the cloud provider throw an error instead of handling it in the Wing SDK.

File: libs/wingsdk/src/target-tf-azure/bucket.ts
Comment (Chriscbr) [reply libs/wingsdk/src/target-tf-azure/bucket.ts:106]: No restrictions that I'm aware of

File: libs/wingsdk/src/cloud/bucket.ts
Comment (Chriscbr) [reply libs/wingsdk/src/cloud/bucket.ts:45]: I've added a comment to the description 👍

**Review by marciocadev — APPROVED** [review]

**Comment by Chriscbr** [comment]
GitHub says I still need an approval from @winglang/maintainers

**Review by staycoolcall911 — APPROVED** [review]

**Pull request merged**
